### ISSUE TITLE
🧭 Organization switcher with procedural icons

### ISSUE THREAD

**Issue by richardolsson** [zetkin/app.zetkin.org#1372]
## Description
A user should be able to see a list of all of their organizations in the main nav organization switcher.

See epic #1370 for context.

## Relevant Job Stories
None

## Prerequisites
Don't start this until #1371 is complete.

## Requirements
- [x] When the menu is in expanded mode, clicking the little downards pointing arrow next to the organization name should expand the top section downward with an animation, to cover the entire menu
- [x] When the organization switcher is expanded, the arrow should be replaced with an upward facing arrow, clicking which will close the switcher again
- [x] If the user expands the switcher before all organizations have been loaded, display a spinner
- [x] All of the user's organizations (in which they have a `role`) should be represented in a tree
- [x] If there is only one top-level organization, it should be expanded by default
- [x] Every organization should use ProceduralColorIcon
- [x] The currently selected organization should be bold
- [x] Hovering an organization should show a slight background tint
- [x] Clicking an organization should navigate to `/organize/{orgId}`

### Retrieving organizations
Organizations can be retrieved using the `/api/users/me/memberships` resource.

* Only organizations in which the user has a `role` should be included in this menu
* An organization is considered top-level (in this context) if the organization has no parent, or if the user does not have a role in that parent

The organization that is currently considered selected is the one whose ID is in the URL, which can be retrieved using `const { orgId } = useNumericRouteParams()`.

### Tree rendering
Maybe use the MUI [`TreeView` component](https://mui.com/material-ui/react-tree-view/)?

### Icon colors - SOLVED
The color of the icons should be procedurally selected. In the API, this is done for personal avatars using an algorithm that uses the ID as a seed for the [`random-seed`](https://npmjs.com/package/random-seed) PRNG:

```ts
const rand = randomSeed.create(id.toString());
const bgR = rand(256);
const bgG = rand(256);
const bgB = rand(256);
const fgR = 255, fgG = 255, fgB = 255;

// Use black text on light backgrounds (when color component
// average is greater than 180).
const bgAvg = (bgR + bgG + bgB) / 3.0;
if (bgAvg > 180) {
  fgR = fgG = fgB = 0;
}
```

## Design specifications
Ignore the "Recent organizations" which is tracked in a separate issue.

![image](https://github.com/zetkin/app.zetkin.org/assets/550212/3558abd3-2740-4d9b-9e4d-3bf55c03f799)

## Open questions
None
